WoWsShipBuilder.Web/Pages/Charts.razor

```diff
@@ -149,365 +150,310 @@
+            string mm = Localizer.GetAppLocalization(nameof(Translation.Unit_MM)).Localization;
+            string degree = Localizer.GetAppLocalization(nameof(Translation.Unit_Degree)).Localization;
+            string mps = Localizer.GetAppLocalization(nameof(Translation.Unit_MPS)).Localization;
+
+            
+            await module.InvokeVoidAsync("SetupGlobalChartConfig");
+            //CreateChart(chartId, title, xLabel, yLabel, xUnit, yUnit)
+            await module.InvokeVoidAsync("CreateChart", HorizontalDispersionId, Localizer.GetAppLocalization(nameof(Translation.DispersionGraphWindow_Horizontal)).Localization + " " + Localizer.GetAppLocalization(nameof(Translation.ShipStats_Dispersion)).Localization, rangeString, Localizer.GetAppLocalization(nameof(Translation.ShipStats_Dispersion)).Localization, km, m);
+            await module.InvokeVoidAsync("CreateChart", VerticalDispersionId, Localizer.GetAppLocalization(nameof(Translation.DispersionGraphWindow_Vertical)).Localization + " " + Localizer.GetAppLocalization(nameof(Translation.ShipStats_Dispersion)).Localization, rangeString, Localizer.GetAppLocalization(nameof(Translation.ShipStats_Dispersion)).Localization, km, m);
+            await module.InvokeVoidAsync("CreateChart", TrajectoryId, Localizer.GetAppLocalization(nameof(Translation.DispersionGraphWindow_ShellsPath)).Localization, rangeString, Localizer.GetAppLocalization(Translation.ChartsWeb_Height).Localization, km, m);
+            await module.InvokeVoidAsync("CreateChart", PenetrationId, Localizer.GetAppLocalization(nameof(Translation.ShipStats_Penetration)).Localization, rangeString, Localizer.GetAppLocalization(nameof(Translation.ShipStats_Penetration)).Localization, km, mm);
+            await module.InvokeVoidAsync("CreateChart", FlightTimeId, Localizer.GetAppLocalization(nameof(Translation.DispersionGraphWindow_FlightTime)).Localization, rangeString, Localizer.GetAppLocalization(nameof(Translation.DispersionGraphWindow_FlightTime)).Localization, km, s);
+            await module.InvokeVoidAsync("CreateChart", ImpactVelocityId, Localizer.GetAppLocalization(nameof(Translation.DispersionGraphWindow_ImpactVelocity)).Localization, rangeString, Localizer.GetAppLocalization(nameof(Translation.DispersionGraphWindow_ImpactVelocity)).Localization, km, mps);
+            await module.InvokeVoidAsync("CreateChart", ImpactAngleId, Localizer.GetAppLocalization(nameof(Translation.DispersionGraphWindow_ImpactAngle)).Localization, rangeString, Localizer.GetAppLocalization(nameof(Translation.DispersionGraphWindow_ImpactAngle)).Localization, km, degree);
+            await ChangeSuggestedMax(TrajectoryId, range / 4 * 1000);
+        }
     }
-
-    private async Task UpdateCharts(List<ShellSelectionWrapper> selectedShips)
+    
+    private async Task UpdateShipsBatch(List<ShellSelectionWrapper> shipList)
     {
+        List<string> chartIds = GetChartsIdList();
+        List<List<IEnumerable<Point>>> data =  new();
```

Comment:
why this structure? Is a triple-nested collection really necessary here?

Reply:
Yes, because it lowers the amount of JSInterop exchange. i can put a comment with an explanation in the code too, but basically: IEnumerable<Point> is a single dataset. The middle List contains all datasets of a certain chart. The outermost List contains the list of datasets of each chart.